coil-base/src/main/java/coil/decode/ImageSource.kt

```diff
@@ -172,10 +172,17 @@ sealed class ImageSource : Closeable {
 }
 
 /**
- * Metadata containing the [fileName] of an Android asset.
+ * Metadata containing the [filePath] of an Android asset.
  */
 @ExperimentalCoilApi
-class AssetMetadata(val fileName: String) : ImageSource.Metadata()
+class AssetMetadata(val filePath: String) : ImageSource.Metadata() {
+
+    @Deprecated(
+        message = "Migrate to filePath as it supports assets inside subfolders.",
+        level = DeprecationLevel.ERROR
+    )
+    val fileName: String get() = filePath.substringAfter('/')
```

Comment:
Shouldn't this be `substringAfterLast`? If `filePath` has multiple subfolders, then `substringAfter` will only 'skip' the first.

Practically speaking, I guess it may not matter too much. Anyone who was using `filePath` before had their file located in the root of the `assets` folder. And for those files `substringAfter` and `substringAfterLast` will yield the same result. 🤷‍♂️

Reply:
Great point! Fixed here: https://github.com/coil-kt/coil/pull/1493